src/Parsers/ParserExplainQuery.cpp

```diff
@@ -0,0 +1,69 @@
+#include <Parsers/ParserExplainQuery.h>
+#include <Parsers/ASTExplainQuery.h>
+#include <Parsers/CommonParsers.h>
+#include <Parsers/ParserSelectWithUnionQuery.h>
+#include <Parsers/ParserSetQuery.h>
+
+namespace DB
+{
+
+bool ParserExplainQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    ASTExplainQuery::ExplainKind kind;
+    bool old_syntax = false;
+
+    ParserKeyword s_ast("AST");
+    ParserKeyword s_analyze("ANALYZE");
+    ParserKeyword s_explain("EXPLAIN");
+    ParserKeyword s_syntax("SYNTAX");
+    ParserKeyword s_pipeline("PIPELINE");
+
+    if (enable_debug_queries && s_ast.ignore(pos, expected))
+    {
+        old_syntax = true;
+        kind = ASTExplainQuery::ExplainKind::ParsedAST;
+    }
+    else if (enable_debug_queries && s_analyze.ignore(pos, expected))
+    {
+        old_syntax = true;
+        kind = ASTExplainQuery::ExplainKind::AnalyzedSyntax;
+    }
+    else if (s_explain.ignore(pos, expected))
+    {
+        kind = ASTExplainQuery::QueryPlan;
+
+        if (s_ast.ignore(pos, expected))
+            kind = ASTExplainQuery::ExplainKind::ParsedAST;
+        else if (s_syntax.ignore(pos, expected))
+            kind = ASTExplainQuery::ExplainKind::AnalyzedSyntax;
+        else if (s_pipeline.ignore(pos, expected))
+            kind = ASTExplainQuery::ExplainKind::QueryPipeline;
```

Comment:
>Usage:
>EXPLAIN [AST | SYNTAX | PLAN | PIPELINE] [setting = value, ...] SELECT ... [FORMAT ...]

Looks like `PLAN` is not supported during parsing, or it wasn't not intended to be supported?

Reply:
I've forgotten about it indeed.